### PR TITLE
fix(fdc3) - Refactored DesktopAgent queries to find the appropriate intents via raiseIntent, findIntent, findIntentsByContext call. Removed unnecessary codes, tests, code cleanup

### PR DESCRIPTION
**Pull request by lilla28**
Changes:
- refactored the intent queries (within backend's RaiseIntent, FindIntent, FindIntentsByContext handlers)
- removed unnecessary code snippets and tests based on: https://github.com/morganstanley/ComposeUI/commit/57ad7daab2f8e28d63b335562b7f8b2bdf20fe79
- fixed typos within classes, descriptions, test utils (sample appdirectory)
- removed unnecessary property from RaiseIntentRequest classes.
image

### CODE REVIEW

File: src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs [hunk 1]
```diff
@@ -32,6 +32,8 @@
 using Icon = MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol.Icon;
 using IntentMetadata = Finos.Fdc3.AppDirectory.IntentMetadata;
 using Screenshot = MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol.Screenshot;
+using MorganStanley.ComposeUI.Fdc3.DesktopAgent.Protocol;
```
Comment (ztanczos) [inline src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs:35]: Unused using statements

File: src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs [hunk 2]
```diff
@@ -124,26 +126,40 @@ public async ValueTask<FindIntentResponse> FindIntent(FindIntentRequest? request
             return FindIntentResponse.Failure(ResolveError.IntentDeliveryFailed);
         }
 
-        //This function returns null, if the app could not be accepted based on the intent (required), context (optional in request), resultType (optional in request)
-        //else for consistency it will return a single element array containing the intentMetadata which is allowed by the request.
-        Func<Fdc3App, IEnumerable<IntentMetadata>?> selector = (fdc3App) =>
+        var appIntents = new Dictionary<string, AppIntent>();
+
+        void FindIntentQuery(Fdc3App fdc3App, string? instanceId)
```
Comment (ztanczos) [inline src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs:131]: Why did you change the return type to `void`? To me a *Query returns something. Similarly the `Get**` methods below should return something, or they should have a different name.

Comment (ztanczos) [reply src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs:131]: But why? If I look at a method which is called `GetSomething()` I'd expect it to return `Something`, and not cause any other side effects.

Comment (lilla28) [reply src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs:131]: Actually I thought that it is more cleaner, but I can revert that back

**Review by ztanczos — APPROVED** [review]

**Review by fhubi — APPROVED** [review]
